lib/minidoc/duplicate_key.rb

```diff
@@ -1,12 +1,9 @@
-class Minidoc::DuplicateKey < Mongo::OperationFailure
+class Minidoc::DuplicateKey < StandardError
   DUPLICATE_KEY_ERROR_CODE = 11000
 
-  def self.duplicate_key_exception(exception)
-    if exception.respond_to?(:error_code) && exception.error_code == DUPLICATE_KEY_ERROR_CODE
-      new(exception.message, exception.error_code, exception.result)
-    else
-      nil
+  def self.duplicate_key_exception(ex)
+    if ex.is_a?(Mongo::Error::OperationFailure) && ex.message.end_with?("(#{DUPLICATE_KEY_ERROR_CODE})")
```

Comment:
Ah we have these already:

```ruby
  describe ".rescue_duplicate_key_errors" do
    it "returns the value of the block when there are no errors" do
      user = User.create!
      result = Minidoc.rescue_duplicate_key_errors { User.create!(name: "two") }
      expect(result.name).to eq "two"
    end

    it "returns false when a duplicate key error occurs" do
      user = User.create!
      result = Minidoc.rescue_duplicate_key_errors do
        User.create!(:_id => user.id, name: "two")
      end
      expect(result).to eq false
    end
  end
```

Given that that spec passes in all the versions, I think it's ok